### PR TITLE
feat(anytracker): `unlock-premium` patch

### PR DESCRIPTION
**Pull request by oSumAtrIX**
## About

Adds back `unlock-premium` patch.

RE: https://github.com/revanced/revanced-patches/pull/1094

## Issue until opening the pull request

After patching, signing the apk is not possible anymore:

```
com.android.apksig.apk.ApkFormatException: Malformed ZIP entry: LICENSE_OFL
	at com.android.apksig.ApkSigner.sign(ApkSigner.java:359)
	at com.android.apksig.ApkSigner.sign(ApkSigner.java:204)
	at app.revanced.utils.signing.Signer.signApk(Signer.kt:77)
	at app.revanced.cli.signing.Signing.sign(Signing.kt:10)
	at app.revanced.cli.command.MainCommand.run(MainCommand.kt:168)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at app.revanced.cli.main.MainKt.main(Main.kt:7)
Caused by: com.android.apksig.zip.ZipFormatException: CRC-32 mismatch between Local File Header and Central Directory for entry LICENSE_OFL. LFH: 4005374435, CD: 2307069120
	at com.android.apksig.internal.zip.LocalFileRecord.getRecord(LocalFileRecord.java:191)
	at com.android.apksig.internal.zip.LocalFileRecord.getRecord(LocalFileRecord.java:127)
	at com.android.apksig.ApkSigner.sign(ApkSigner.java:356)
	... 13 more
```

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
#1076 by @FineFindus.

File: src/main/kotlin/app/revanced/patches/anytracker/misc/premium/annotations/UnlockPremiumCompatibility.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package app.revanced.patches.anytracker.misc.premium.annotations
+
+import app.revanced.patcher.annotation.Compatibility
+import app.revanced.patcher.annotation.Package
+
+@Compatibility([Package("com.shervinkoushan.anyTracker")])
```
Comment (FineFindus) [inline src/main/kotlin/app/revanced/patches/anytracker/misc/premium/annotations/UnlockPremiumCompatibility.kt:6]: I'm guessing the dev will continue to change his app, would it be a good idea to add a version constraint?

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/anytracker/misc/premium/annotations/UnlockPremiumCompatibility.kt:6]: I believe we can add the constraint when necessary, for now the fingerprint should work for future versions as well.

**Comment by FineFindus** [comment]
Apply the patch worked for me, but the patch itself fails.
```
5506  5531 E AndroidRuntime:java.lang.IncompatibleClassChangeError: Class 'java.lang.Boolean' does not implement interface 'kotlinx.coroutines.flow.Flow' in call to 'java.lang.Object kotlinx.coroutines.flow.Flow.collect(kotlinx.coroutines.flow.FlowCollector, kotlin.coroutines.Continuation)' (declaration of 'com.shervinkoushan.anyTracker.billing.BillingDataSource$c' appears in /data/app/~~e6CAM7-rUTJ0bD0trDyX-Q==/com.shervinkoushan.anyTracker-SLa8R_yKT2Fe4lNsgs36Fg==/base.apk)
5506  5531 E AndroidRuntime: 	at com.shervinkoushan.anyTracker.billing.BillingDataSource$c.collect(SourceFile:1)
```

**Comment by oSumAtrIX** [comment]
Weird, signing fails for me. Wild guess is that I am using the wrong Java 17 distribution when I recently updated my IDE. Based on the exception you sent, the wrong register is probably returned.

**Comment by FineFindus** [comment]
I opened the #1175, which fixes the patch.

**Comment by oSumAtrIX** [comment]
Thanks @FineFindus.

**Pull request merged**